timewarp_gl/plugin.cpp

```diff
@@ -433,69 +438,109 @@ class timewarp_gl : public threadloop {
 
-
 		// Config distortion uv2 vbo
 		glGenBuffers(1, &distortion_uv2_vbo);
 		glBindBuffer(GL_ARRAY_BUFFER, distortion_uv2_vbo);
-		glBufferData(GL_ARRAY_BUFFER, HMD::NUM_EYES * (num_distortion_vertices * 2) * sizeof(GLfloat), distortion_uv2, GL_STATIC_DRAW);
+
+        HMD::uv_coord_t* const distortion_uv2_data = distortion_uv2.data();
+        assert(distortion_uv2_data != nullptr && "Timewarp allocation should not fail");
+		glBufferData(GL_ARRAY_BUFFER, num_elems_pos_uv * sizeof(HMD::uv_coord_t), distortion_uv2_data, GL_STATIC_DRAW);
+
 		glVertexAttribPointer(distortion_uv2_attr, 2, GL_FLOAT, GL_FALSE, 0, 0);
 		//glEnableVertexAttribArray(distortion_uv2_attr);
 
-
 		// Config distortion mesh indices vbo
 		glGenBuffers(1, &distortion_indices_vbo);
 		glBindBuffer(GL_ELEMENT_ARRAY_BUFFER, distortion_indices_vbo);
-		glBufferData(GL_ELEMENT_ARRAY_BUFFER, num_distortion_indices * sizeof(GLuint), distortion_indices, GL_STATIC_DRAW);
 
+        GLuint* const distortion_indices_data = distortion_indices.data();
+		assert(distortion_indices_data != nullptr && "Timewarp allocation should not fail");
+		glBufferData(GL_ELEMENT_ARRAY_BUFFER, num_distortion_indices * sizeof(GLuint), distortion_indices_data, GL_STATIC_DRAW);
 
-                // Config PBO for texture image collection
-                glGenBuffers(1, &PBO_buffer);
-                glBindBuffer(GL_PIXEL_PACK_BUFFER, PBO_buffer);
-                glBufferData(GL_PIXEL_PACK_BUFFER, SCREEN_WIDTH * SCREEN_HEIGHT * 3, 0, GL_STREAM_DRAW);
+        // Config PBO for texture image collection
+        glGenBuffers(1, &PBO_buffer);
+        glBindBuffer(GL_PIXEL_PACK_BUFFER, PBO_buffer);
+        glBufferData(GL_PIXEL_PACK_BUFFER, SCREEN_WIDTH * SCREEN_HEIGHT * 3, 0, GL_STREAM_DRAW);
 
+        RAC_ERRNO_MSG("timewarp_gl before glXMakeCurrent");
 
-		glXMakeCurrent(xwin->dpy, None, NULL);
+        [[maybe_unused]] const bool gl_result_1 = static_cast<bool>(glXMakeCurrent(xwin->dpy, None, nullptr));
+		assert(gl_result_1 && "glXMakeCurrent should not fail");
 	}
 
-	virtual void warp([[maybe_unused]] float time) {
-		glXMakeCurrent(xwin->dpy, xwin->win, xwin->glc);
+	virtual void warp([[maybe_unused]] std::chrono::system_clock::time_point time) {
+		assert(errno == 0 && "Errno should not be set at start of warp");
+
+        [[maybe_unused]] const bool gl_result = static_cast<bool>(glXMakeCurrent(xwin->dpy, xwin->win, xwin->glc));
+		assert(gl_result && "glXMakeCurrent should not fail");
 
 		glBindFramebuffer(GL_FRAMEBUFFER,0);
 		glViewport(0, 0, SCREEN_WIDTH, SCREEN_HEIGHT);
 		glClearColor(0, 0, 0, 0);
-    	glClear(GL_COLOR_BUFFER_BIT | GL_DEPTH_BUFFER_BIT | GL_STENCIL_BUFFER_BIT);
+
+        glClear(GL_COLOR_BUFFER_BIT | GL_DEPTH_BUFFER_BIT | GL_STENCIL_BUFFER_BIT);
+        RAC_ERRNO_MSG("timewarp_gl after glClear");
+        /// RAC is triggered by glClear consistently. Checking for errors here to be safe.
+        const GLenum glClearError = glGetError();
```

Comment:
Why is this necessary? Won't GL errors get reported via the debug callback?

Reply:
We can remove it. This one was just for testing/abundance of caution.